mypy_django_plugin/transformers/managers.py

```diff
@@ -115,6 +121,66 @@ def get_funcdef_type(definition: Union[FuncBase, Decorator, None]) -> Optional[P
     )
 
 
+def _get_base_containing_method(queryset_info: TypeInfo, method_name: str) -> Instance | None:
+    for mro in queryset_info.mro:
+        for base in mro.bases:
+            if method_name in base.type.names:
+                return base
+    return None
+
+
+def _manage_type_var(ret_type: MypyType, base: Instance) -> MypyType:
+    """
+    Attempts to manage the concrete type corresponding to a `TypeVarType` in a queryset's base types.
+    If nothing is found, keep it unchanged.
+
+    Example:
+
+    T = TypeVar("T")
+    for def example_list_dict(self) -> list[dict[str, T]]
+
+    Tries to find the corresponding `TypeVar` of `T` if it has been set.
+
+    """
+    if isinstance(ret_type, TypeVarType):
+        return _find_type_var(base, ret_type) or ret_type
+    elif isinstance(ret_type, Instance):
+        # Since it is an instance, recursively find the type var for all its args.
+        ret_type.args = tuple(_manage_type_var(item, base) for item in ret_type.args)
+    if isinstance(ret_type, ProperType) and hasattr(ret_type, "item"):
+        # For example TypeType has an item. find the type_var for this item
+        ret_type.item = _manage_type_var(ret_type.item, base)
+    if isinstance(ret_type, ProperType) and hasattr(ret_type, "items"):
+        # For example TypeList has items. find recursively type_var for its items
+        ret_type.items = [_manage_type_var(item, base) for item in ret_type.items]
+    return ret_type
+
+
+def _find_type_var(base: Instance, ret_type: TypeVarType) -> Optional[MypyType]:
+    """
+    Attempts to find the concrete type corresponding to a TypeVarType in a queryset's base types.
+
+    Example:
+    Suppose queryset_info is based on a generic class `QuerySet[T]` and ret_type corresponds
+    to `T`, then this function will return the concrete type that `T` was instantiated with.
+    """
+    for i, type_var in enumerate(base.type.defn.type_vars):
```

Comment:
```suggestion
    for i, type_var in enumerate(base.type.type_vars):
```

No need to use the definition here, I think.

Reply:
base.type.type_vars are strings. I can get the fullname with `base.type.defn.type_vars`